cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/LockDao.java

```diff
@@ -0,0 +1,313 @@
+                    projectId.getName(), projectId.getOfficeId());
+            return catalogResults.stream().map(LockDao::catMap).collect(toList());
+        });
+    }
+
+    public Lock retrieveLock(CwmsId lockId, UnitSystem units) {
+        if (units == null) {
+            units = UnitSystem.SI;
+        }
+        UnitSystem unitSystemFinal = units;
+        return connectionResult(dsl, c -> {
+            setOffice(c, lockId.getOfficeId());
+            LocationsDao locationsDao = new LocationsDaoImpl(dsl);
+            Record dbRecord = DSL.using(c)
+                .select(asterisk())
+                .from(view)
+                .where(view.LOCK_ID.eq(lockId.getName()).and(view.DB_OFFICE_ID.eq(lockId.getOfficeId()))
+                        .and(view.UNIT_SYSTEM.equalIgnoreCase(unitSystemFinal.name())))
+                .fetchOne();
+            if (dbRecord == null) {
+                throw new NotFoundException("Lock not found: " + lockId);
+            }
+
+            Long chamberTypeCode = dbRecord.get(view.CHAMBER_LOCATION_DESCRIPTION_CODE);
+            Table<?> chamberTable = table("CWMS_20.AT_LOCK_GATE_TYPE");
+            Table<?> officeTable = table("CWMS_20.CWMS_OFFICE");
+            SelectConditionStep<Record> chamberQuery = DSL.using(c).select(asterisk())
+                    .from(chamberTable
+                            .join(officeTable)
+                            .on(field(name(chamberTable.getQualifiedName().unquotedName().toString(),
+                                        "DB_OFFICE_CODE").unquotedName())
+                                    .eq(field(name(officeTable.getQualifiedName().unquotedName().toString(),
+                                            "OFFICE_CODE").unquotedName()))))
+                    .where(field(name(chamberTable.getQualifiedName().unquotedName().toString(),
+                                "CHAMBER_TYPE_CODE").unquotedName())
+                            .eq(val(chamberTypeCode)));
+            Record chamberResult = chamberQuery.fetchOne();
+            Location lockLocation = locationsDao.getLocation(dbRecord.get(view.LOCK_ID),
+                    "SI", dbRecord.get(view.DB_OFFICE_ID));
+            if (chamberResult == null) {
+                return map(dbRecord, lockLocation, null);
+            }
+            LookupType chamber = new LookupType.Builder()
+                    .withOfficeId(chamberResult.get("OFFICE_ID", String.class))
+                    .withActive(chamberResult.get("CHAMBER_TYPE_ACTIVE", Boolean.class))
+                    .withTooltip(chamberResult.get("CHAMBER_TYPE_TOOLTIP", String.class))
+                    .withDisplayValue(chamberResult.get("CHAMBER_TYPE_DISPLAY_VALUE", String.class))
+                    .build();
+
+            Lock retVal =  map(dbRecord, lockLocation, chamber);
+            return unitConvert(retVal, unitSystemFinal);
+        });
+    }
+
+    private Lock unitConvert(Lock lock, UnitSystem unitSystem) {
```

Comment:
Added this unit conversion to fix a DB bug where the view provides the minimum draft in the default DB units and not the units specified when storing a lock.

Reply:
This should be resolved with a db patch now